src/mnemonic.rs

```diff
@@ -0,0 +1,41 @@
+/// FuelMnemonic is a simple mnemonic phrase generator.
+pub struct FuelMnemonic;
+
+#[cfg(feature = "std")]
+mod use_std {
+    use super::FuelMnemonic;
+    use crate::{Error, SecretKey};
+    use coins_bip32::path::DerivationPath;
+    use coins_bip39::{English, Mnemonic};
+    #[cfg(feature = "random")]
+    use rand::Rng;
+    use std::str::FromStr;
+
+    type W = English;
+
+    impl SecretKey {
+        /// Generate a new secret key from a mnemonic phrase and its derivation path.
+        /// The derivation path is a list of integers, each representing a child index.
+        pub fn new_from_mnemonic_phrase_with_path(phrase: &str, path: &str) -> Result<Self, Error> {
```

Comment:
It is possible to build both `DerivationPath` and `Mnemonic` other than using a path and phrase, respectively.

Apart from this ad-hoc method, shouldn't we make available also a `new_from_mnemonic` that will simply take these two and give the user the possibility to decide how he wants to generate the mnemonic? Or do we intentionally want to restrict the user from doing that? If yes, it might be good to add a comment explaining why we don't want the user to create his own mnemonic

Reply:
> Or do we intentionally want to restrict the user from doing that?

Nope, that wasn't the intention. I can't see any reason for placing this restriction. 

> shouldn't we make available also a new_from_mnemonic that will simply take these two and give the user the possibility to decide how he wants to generate the mnemonic?

By "take these two [...]", you mean having a `new_from_mnemonic(d: DerivationPath, m: Mnemonic)`? If that's the case, that's a great idea!

Reply:
Yes, that's perfect